README.md

```diff
@@ -22,23 +22,38 @@ We use the [ruff](https://docs.astral.sh/ruff/) tool to format our code and ensu
 without human error. Before submitting a PR, please install the dev requirements for the project and ensure your changes
 do not fail ruff executions.
 
+
+### Quick developer setup
+
+When beginning your work, run `source ./dev-setup.sh`. This script ensures it has a valid development environment and
+sets up pre-commit hooks to validate and format Python code files.
+
+
+### Manual dev tool setup and workflow steps
+
+Create a branch off of the latest commits to `main`:
+
 ```bash
-# Set up a working dev environment
+git checkout main
+git pull
 git checkout -b your-branch
+```
+
+Set up a working dev environment:
+
+```bash
 virtualenv venv
 . ./venv/bin/activate
 pip install .[dev]
-
-# Make whatever edits you need now.
-# vim tb_pulumi/iam.py
-
-# Format files so they adhere to our standards
```

Comment:
You might keep this section, so that folks know how to run this tools manually? Maybe just add to a different spot?